fastplotlib/plot.py

```diff
@@ -103,4 +104,57 @@ def show(self, autoscale: bool = True):
         if autoscale:
             self.auto_scale(maintain_aspect=True, zoom=0.95)
 
-        return self.canvas
+        if toolbar:
+            tools = ToolBar(self).toolbar
+            return VBox([self.canvas, tools])
+        else:
+            return self.canvas
+
+
+class ToolBar:
+    def __init__(self,
+                 plot: Plot):
+        """
+        Basic toolbar for a Plot instance.
+
+        Parameters
+        ----------
+        plot: encapsulated plot instance that will be manipulated using the toolbar buttons
+        """
+        self.plot = plot
+
+        self.autoscale_button = Button(value=False, disabled=False, icon='expand-arrows-alt',
+                                       layout=Layout(width='auto'), tooltip='Auto-scale the camera w.r.t to the scene')
+        self.center_scene_button = Button(value=False, disabled=False, icon='compress-arrows-alt',
+                                          layout=Layout(width='auto'), tooltip='Auto-center the scene, does not scale')
+        self.panzoom_controller_button = ToggleButton(value=True, disabled=False, icon='hand-pointer',
+                                                      layout=Layout(width='auto'), tooltip='Toggle panzoom controller')
+        self.maintain_aspect_button = ToggleButton(value=True, disabled=False, description="1:1",
+                                                   layout=Layout(width='auto'),
+                                                   tooltip='Maintain camera aspect ratio for all dims')
+        self.maintain_aspect_button.style.font_weight = "bold"
+
+        self._widget = HBox([self.autoscale_button,
+                             self.center_scene_button,
+                             self.panzoom_controller_button,
+                             self.maintain_aspect_button])
+
+        def auto_scale(obj):
```

Comment:
Make all of these regular methods. I don't like nesting functions unless it's necessary